### ISSUE TITLE
problem with optgroup

### ISSUE THREAD

**Issue by XarchencoMO** [lekoala/bootstrap5-autocomplete#14]
I have a input with datalist, then contains groped options, and it's doesn't work
P.S. I am using Laravel blade templates for adding opt groups and options. 
`<div>
    <script type="module">
        import Autocomplete from "https://cdn.jsdelivr.net/gh/lekoala/bootstrap5-autocomplete@master/autocomplete.js";
        const opts = {
            onSelectItem: console.log,
        };
        Autocomplete.init("input.autocomplete", {
            valueField: "id",
            labelField: "title",
            highlightTyped: true,
            onSelectItem: console.log,
        });
        // We can use regular objects as source and customize label
        new Autocomplete(document.getElementById("autocompleteDatalist"), opts);
    </script>
    <style>
        /* highlightTyped use mark */
        .autocomplete-menu mark {
            text-decoration: underline;
            background: none;
            color: currentColor;
            padding: 0;
        }
    </style>
    <label for="autocompleteInputZero" class="form-label">Type something (zero threshold)</label>
    <input data-datalist="list-estates" type="text" class="form-control autocomplete" id="autocompleteInputZero" data-suggestions-threshold="0" placeholder="Type something"/>
    <datalist id="list-estates">
        @foreach($groupedItems as $organizationId => $items)
            <optgroup label="{{$organizationId}}">
                @foreach($items as $item)
                    <option value="{{$item->id}}">{{$item->object_branch_name}}</option>
                @endforeach
            </optgroup>
        @endforeach
    </datalist>
</div>
`

**Comment by lekoala** [comment]
yes currently option group are not supported (yet, maybe i will spend some time on it but i haven't seen the real need yet)
https://github.com/lekoala/bootstrap5-autocomplete/issues/7

it wouldn't be too hard to add, but i'm not sure it's really worth it. this is an autocomplete library, not a replacement for a select box. datalist, for examples, don't support optgroup either

in my other library
https://github.com/lekoala/bootstrap5-tags
i've added support for optgroups, it's more like a real select
